sunbird-es-utils/src/main/java/org/sunbird/common/ElasticSearchRestHighImpl.java

```diff
@@ -0,0 +1,1033 @@
+                mappedIndexAndType.get(JsonKey.INDEX),
+                mappedIndexAndType.get(JsonKey.TYPE),
+                identifier)
+            .source(data);
+
+    ActionListener<IndexResponse> listener =
+        new ActionListener<IndexResponse>() {
+          @Override
+          public void onResponse(IndexResponse indexResponse) {
+            ProjectLogger.log(
+                "ElasticSearchUtilRest createData Success for type : "
+                    + type
+                    + ", identifier :"
+                    + identifier,
+                LoggerEnum.INFO);
+
+            promise.success(indexResponse.getId());
+            ProjectLogger.log(
+                "ElasticSearchUtilRest createData method end at =="
+                    + System.currentTimeMillis()
+                    + " for Type "
+                    + type
+                    + " ,Total time elapsed = "
+                    + calculateEndTime(startTime),
+                LoggerEnum.PERF_LOG);
+          }
+
+          @Override
+          public void onFailure(Exception e) {
+            promise.failure(e);
+            ProjectLogger.log(
+                "Error while saving " + type + " id : " + identifier + " with error :" + e,
+                LoggerEnum.ERROR.name());
+            ProjectLogger.log(
+                "ElasticSearchUtilRest createData method end at =="
+                    + System.currentTimeMillis()
+                    + " for Type "
+                    + type
+                    + " ,Total time elapsed = "
+                    + calculateEndTime(startTime),
+                LoggerEnum.PERF_LOG);
+          }
+        };
+
+    ConnectionManager.getRestClient().indexAsync(indexRequest, listener);
+
+    return promise.future();
+  }
+
+  @Override
+  public Future<Boolean> updateData(
+      String index, String type, String identifier, Map<String, Object> data) {
+    long startTime = System.currentTimeMillis();
+    ProjectLogger.log(
+        "ElasticSearchUtilRest updateData method started at ==" + startTime + " for Type " + type,
+        LoggerEnum.PERF_LOG);
+    Promise<Boolean> promise = Futures.promise();
+    ;
+
+    if (!StringUtils.isBlank(index)
+        && !StringUtils.isBlank(type)
+        && !StringUtils.isBlank(identifier)
+        && data != null) {
+      Map<String, String> mappedIndexAndType =
+          ElasticSearchHelper.getMappedIndexAndType(index, type);
+      UpdateRequest updateRequest =
+          new UpdateRequest(
+                  mappedIndexAndType.get(JsonKey.INDEX),
+                  mappedIndexAndType.get(JsonKey.TYPE),
+                  identifier)
+              .doc(data);
+
+      ActionListener<UpdateResponse> listener =
+          new ActionListener<UpdateResponse>() {
+            @Override
+            public void onResponse(UpdateResponse updateResponse) {
+              promise.success(true);
+              if (updateResponse.getResult() == DocWriteResponse.Result.CREATED) {
```

Comment:
Just write a logger that will indicate Operation type done by ES. for end user it should be success always.
Remove all if, elseif and elseif loop.

Reply:
Done